### ISSUE TITLE
adding TimedStopping Callback (like EarlyStopping but for time)

### ISSUE THREAD

**Issue by shun-lin** [tensorflow/addons#750]
**Describe the feature and the current behavior/state.**

TimedStopping callback would be a tf callback that takes in a parameter from users (like say number of seconds) and stops the training when that much time passed since the start of the training, much like EarlyStopping callback but instead of monitoring a metrics, the callback would trigger end of training based on time.

This may be useful for people who want to train a model for a certain amount of time. Personally, sometimes I would want to train models for a certain amount of time (and there are codes after training ends so CTRL+C is not ideal) and I often find myself using pen and pencil to calculate how many epochs I would want to run. I hope people may find this useful as well.

**Relevant information**
- Are you willing to contribute it (yes/no): Yes
- Are you willing to maintain it going forward? (yes/no): Yes
- Is there a relevant academic paper? (if so, where): No
- Is there already an implementation in another framework? (if so, where): No
- Was it part of tf.contrib? (if so, where): No, but an issue was open for this callback but the author never implemented it in tf.contrib.

Relevant Issues (old, those are from 2017/2016):
https://github.com/keras-team/keras/issues/1625
https://github.com/keras-team/keras-contrib/issues/87

**Which API type would this fall under (layer, metric, optimizer, etc.)**
Callbacks.

**Who will benefit with this feature?**
People who wants an EarlyStopping like callback that stops training after a certain amount of time.

**Comment by seanpmorgan** [comment]
Interesting idea. Personally have no issue including this in the repo (given you'll maintain) as it fits nicely in callbacks API. Thanks for your continued contributions!

**Comment by shun-lin** [comment]
thanks @seanpmorgan :)

**Comment by shun-lin** [comment]
merged! See tutorial here https://github.com/tensorflow/addons/blob/master/docs/tutorials/time_stopping.ipynb

Thanks @seanpmorgan
